src/docfx/build/page/BuildPage.cs

```diff
@@ -52,6 +53,25 @@ internal static class BuildPage
             return (errors, output, dependencies.Build());
         }
 
+        private static string GetCanonicalUrl(Document file)
```

Comment:
>GetCanonicalUrl [](start = 30, length = 15)

do we still need the `_op_canonicalUrlPrefix` generating logic in  legacy?

Reply:
no. that will be a separate PR.